nucypher/characters/lawful.py

```diff
@@ -1286,23 +1265,27 @@ def run(self,
             emitter.message(f"Starting services", color='yellow')
 
         if pruning:
-            self.__pruning_task = self._datastore_pruning_task.start(interval=self._pruning_interval, now=True)
+            self.__pruning_task = self._datastore_pruning_task.start(interval=self._pruning_interval, now=eager)
             if emitter:
                 emitter.message(f"✓ Database Pruning", color='green')
 
-        # TODO: Startup node discovery here with the rest of Ursula's services.
-        # if discovery and not self.lonely:
-        #     self.start_learning_loop(now=self._start_learning_now)
-        #     if emitter:
-        #         emitter.message(f"✓ Node Discovery - {self.domain}", color='green')
+        if discovery and not self.lonely:
+            self.start_learning_loop(now=eager)
+            if emitter:
+                emitter.message(f"✓ Node Discovery ({self.domain.capitalize()})", color='green')
 
-        if self._availability_check and availability:
-            self._availability_tracker.start(now=False)  # wait...
+        if self._availability_check or availability:
+            self._availability_tracker.start(now=eager)
             if emitter:
                 emitter.message(f"✓ Availability Checks", color='green')
 
         if worker and not self.federated_only:
-            self.work_tracker.start(commit_now=True)  # requirement_func=self._availability_tracker.status)  # TODO: #2277
+            if block_until_ready:
+                # Sets (staker's) checksum address; Prevent worker startup before bonding
+                self.block_until_ready()
+            self.stakes.checksum_address = self.checksum_address
+            self.stakes.refresh()
+            self.work_tracker.start(commit_now=eager)  # requirement_func=self._availability_tracker.status)  # TODO: #2277
```

Comment:
We seem to be using now the `eager` parameter for several things (availability tracker, work tracker, prunning task)....maybe this can have unintended consequences? I'm not sure what's the expectation of the callers to this constructor.

Reply:
Perhaps it covers too many cases, but I don't know if there's a need for any more granular control of individual services here?